truss/environment_inference/requirements_inference.py

```diff
@@ -52,6 +54,11 @@ def infer_deps(must_include_deps: Set[str] = None) -> Set[str]:
     for m in pkg_resources.working_set:
         if m.project_name in imports and m.project_name not in IGNORED_PACKAGES:
             requirements.add(f"{m.project_name}=={m.version}")
+            if m.project_name in must_include_deps:
+                must_include_deps.remove(m.project_name)
```

Comment:
I'm not sure I follow this code, but must_include_deps is an argument to the function, modifying it inside the function can be confusing and error prone. Also, good to rename `m` to something more descriptive at this point, such short names are only fit for one liners.

Reply:
The issue here is that if a dependency included in `must_include_deps` isn't found in `pkg_candidates` (which is generated from the current frame), those "must include" dependencies aren't included. This change allows for all `must_include_deps` to be included, even if they aren't found in the current frame. Hope that provides some clarity! 

I will refactor to avoid mutating the argument.